Makefile

```diff
@@ -157,8 +159,55 @@ test-unit-go:
 	cd ./relayer && go test -v ./... -race -count=10
 
 .PHONY: test-integration
-test-integration: test-integration-smoke
+test-integration: test-integration-smoke test-integration-contracts test-integration-gauntlet
 
 .PHONY: test-integration-smoke
 test-integration-smoke:
 	ginkgo -v -r --junit-report=tests-smoke-report.xml --keep-going --trace integration-tests/smoke
+
+.PHONY: test-integration-contracts
+# TODO: better network lifecycle setup - requires external network (L1 + L2)
+test-integration-contracts: build-ts env-devnet-hardhat
+	cd examples/contracts/aggregator-consumer/ && \
+		yarn test
+	cd packages-ts/integration-eqlabs-multisig/ && \
+		yarn test
+	cd packages-ts/integration-starkgate/ && \
+		yarn test
+
+.PHONY: test-integration-gauntlet
+# TODO: better network lifecycle setup - tests setup/run their own network (L1 + conflict w/ above if not cleaned up)
+test-integration-gauntlet: build-ts env-devnet-hardhat-down
+	cd packages-ts/starknet-gauntlet/ && \
+		yarn test
+	cd packages-ts/starknet-gauntlet-argent/ && \
+		yarn test
+	cd packages-ts/starknet-gauntlet-cli/ && \
+		yarn test
+	cd packages-ts/starknet-gauntlet-example/ && \
+		yarn test
+	cd packages-ts/starknet-gauntlet-multisig/ && \
+		yarn test
+	cd packages-ts/starknet-gauntlet-ocr2/ && \
+		yarn test
+	cd packages-ts/starknet-gauntlet-oz/ && \
+		yarn test
+	cd packages-ts/starknet-gauntlet-starkgate/ && \
+		yarn test
```

Comment:
Why not rely on the toplevel `yarn test` which uses the Jest config to run all the tests? I personally only use that and ignore the `Makefile`

Reply:
The version before this PR just ran Jest tests via root `package.json` test script.

The problem was not all workspace packages use Jest for testing - our contracts, example project, and 2x integration projects (multisig, starkgate) use hardhat + mocha, so they wouldn't be run.

Then I updated the root `package.json` test script to iterate over all projects and run their test scripts via `"yarn workspaces run test"` - but now the problem is Hardhat setup tests expect external network to be started already, and Gauntlet Jest tests run their own network (Integrated Devnet) on `beforeAll`. This was now a conflict because Integrated Devnet wouldn't start successfully if docker container was already running on the same 5050 port.

So for now I've split the two category of tests as different target, and different CI workflows. Now there is no infra conflict, and all tests are finally ran.

This is enough for this PR I'd say, and the plan for the next one is to restructure how do we run local env in tests as dependencies. Every tests suite should start its own network programatically in test setup, and tear it down when finished. At that point we can just run all tests at once via `"yarn workspaces run test"` and they would be even more isolated.

Reply:
tldr; I consider this a temp workaround, and will try to remove it as test runs are restructured to be idempotent.